### PR TITLE
Use correct app name in warnings, allow using app::backup for restore source

### PR DESCRIPTION
**Pull request by tef**
/cc @uhoh-itsmaciek

### CODE REVIEW

File: lib/heroku/command/pg_backups.rb [hunk 1]
```diff
@@ -26,9 +26,9 @@ def copy
     message = "WARNING: Destructive Action"
     message << "\nThis command will remove all data from #{target.name}"
     message << "\nData from #{source.name} will then be transferred to #{target.name}"
-    message << "\nThis command will affect the app: #{app}"
+    message << "\nThis command will affect the app: #{attachment.app}"
 
-    if confirm_command(app, message)
+    if confirm_command(attachment.app, message)
```
Comment (tef) [inline lib/heroku/command/pg_backups.rb]: I think I might need to fix the logic here.

File: lib/heroku/command/pg_backups.rb [hunk 2]
```diff
@@ -380,16 +384,16 @@ def restore_backup
...
-        abort("No backups. Capture one with `heroku pg:backups capture`.")
+        abort("No backups for #{backup_app}. Capture one with `heroku pg:backups capture`.")
       elsif backup.nil?
-        abort("Backup #{backup_name} not found.")
+        abort("Backup #{backup_name} not found for #{backup_app}.")
       elsif !backup[:succeeded]
-        abort("Backup #{backup_name} did not complete successfully; cannot restore it.")
+        abort("Backup #{backup_name} for #{backup_app} did not complete successfully; cannot restore it.")
       end
       restore_url = backup[:to_url]
```
Comment (tef) [inline lib/heroku/command/pg_backups.rb]: This url should be fine if we retrieve it from another app? Or will we need to resolve it to a public url ?

Comment (uhoh-itsmaciek) [inline lib/heroku/command/pg_backups.rb]: Let's just resolve it so we can tighten the group restrictions.

File: lib/heroku/command/pg_backups.rb (hunk 1 above)
Comment (uhoh-itsmaciek) [inline lib/heroku/command/pg_backups.rb]: Not `target.attachment.app`?

Comment (uhoh-itsmaciek) [inline lib/heroku/command/pg_backups.rb]: And ugh, great catch.

Comment (tef) [inline lib/heroku/command/pg_backups.rb]: the target could have no attachment? would we want a warning then ?

**Comment by tef** [comment]
@uhoh-itsmaciek 

Fixes:

- EU Timezones, Zulu Timezone
- Bad warnings, should always show target database if specified in copy, or restore
- Restoring to a database using source_app::backup
- Make schedule time non optional

I know we should improve the url stuff, but I think this is good to go

File: lib/heroku/command/pg_backups.rb (hunk 1 above)
Comment (uhoh-itsmaciek) [inline lib/heroku/command/pg_backups.rb]: Good call--if you're restoring *from* a Heroku Postgres database to a non-HPG DB, only `name` and `url` fields of the `target` will be valid (it's a `MaybeAttachment`; see below in `resolve_db_or_url`). The

File: lib/heroku/command/pg_backups.rb (hunk 2 above)
Comment (uhoh-itsmaciek) [inline lib/heroku/command/pg_backups.rb]: After meatspace convo, that can wait: this should work.

**Comment by uhoh-itsmaciek** [comment]
Yeah, +1. The only thing is the `confirm_command` warning--I think the code I outlined should work for that.

File: lib/heroku/command/pg_backups.rb (hunk 1 above)
Comment (tef) [inline lib/heroku/command/pg_backups.rb]: Ah, Nice.

**Comment by tef** [comment]
Added, tested. This is good to go

**Comment by dickeyxxx** [comment]
it's got build failures

**Comment by tef** [comment]
@dickeyxxx I missed a : in the output, specs pass locally now.

**Pull request merged**